bears/python/PyImportSortBear.py

```diff
@@ -176,13 +177,36 @@ def run(self, filename, file,
             isort_settings['known_standard_library'] = (
                 known_standard_library_imports)
 
-        sort_imports = SortImports(file_contents=''.join(file),
-                                   **isort_settings)
-        new_file = tuple(sort_imports.output.splitlines(True))
+        if treat_seperated_imports_independently:
+            import_stmts = []
+            tmp = []
+            for lineno, lines in enumerate(file, start=1):
+                if 'import' in lines:
+                    tmp.append((lineno, lines))
+                else:
+                    import_stmts.append(tmp)
+                    tmp = []
 
-        if new_file != tuple(file):
-            diff = Diff.from_string_arrays(file, new_file)
-            yield Result(self,
-                         'Imports can be sorted.',
-                         affected_code=diff.affected_code(filename),
-                         diffs={filename: diff})
+            for units in import_stmts:
+                sort_imports = SortImports(file_contents=''.join([x[1]
+                                                            for x in units]),
+                                           **isort_settings)
+                new_file = tuple(sort_imports.output.splitlines(True))
+
+                if new_file != tuple(file):
+                    diff = Diff.from_string_arrays(file, new_file)
+                    yield Result(self,
+                                 'Imports can be sorted.',
+                                 affected_code=diff.affected_code(filename),
+                                 diffs={filename: diff})
```

Comment:
Would be better if I could not duplicate the code, wrap the reused code in a function maybe :confused: :question: